xarray/backends/api.py

```diff
@@ -397,11 +408,13 @@ def _dataset_from_backend_dataset(
         )
 
     _protect_dataset_variables_inplace(backend_ds, cache)
-    if chunks is None:
-        ds = backend_ds
-    else:
+
+    if create_default_indexes:
+        ds = _maybe_create_default_indexes(backend_ds)
```

Comment:
had to be careful here. `backend_ds = _maybe_create_default_indexes(backend_ds)` would lose the `_close` attribute (because of the copy executed by `assign_coords`)